### PR TITLE
upgrade 2.12 with vendir

### PR DESCRIPTION
**Pull request by mcharriere**
This PR updates linkerd2-app using vendir. 

towards https://github.com/giantswarm/giantswarm/issues/23766 https://github.com/giantswarm/giantswarm/issues/23384

## Checklist

- [ ] Automated test are working (for chart changes)
- [ ] I am confident my changes don't break existing installations (for chart changes)
- [ ] Changelog entry has been added

### CODE REVIEW

**Comment by mcharriere** [comment]
Small guide for reviewers:

You can ignore the files under helm/linkerd2-cni-app/charts and helm/linkerd2-cni-app/templates.
They are coming directly from https://github.com/giantswarm/linkerd2-upstream/tree/giantswarm/main.

The rest is up for review.

As the PR is big enough already, I'll leave the app subfix removal as the next step

File: helm/linkerd2-app/values.yaml [hunk 1]
```diff
@@ -233,9 +261,9 @@ controllerUID: 2103
 # destination configuration
 # set resources for the sp-validator and its linkerd proxy respectively
 # see proxy.resources for details.
-# destinationResources -- CPU and Memory resources required by destination (see `proxy.resources` for sub-fields)
-destinationResources: *controller_resources
-# destinationProxyResources -- CPU and Memory resources required by proxy
+# destinationResources -- CPU, Memory and Ephemeral Storage resources required by destination (see `proxy.resources` for sub-fields)
+# destinationResources:
```
Comment (mcharriere) [inline helm/linkerd2-app/values.yaml:265]: let's not remove the defaults for resources

File: helm/linkerd2-app/values.yaml [hunk 2]
```diff
@@ -53,17 +52,20 @@ cniEnabled: true
 identityTrustAnchorsPEM: |
 # -- Trust domain used for identity
 # @default -- clusterDomain
-identityTrustDomain: *cluster_domain
+identityTrustDomain: ""
```
Comment (mcharriere) [inline helm/linkerd2-app/values.yaml:55]: review this change

File: helm/linkerd2-app/values.yaml [hunk 3]
```diff
@@ -72,13 +74,14 @@ policyController:
     # @default -- linkerdVersion
     version: ""
 
-  # -- The default allow policy to use when no `Server` selects a pod.  One of: "all-authenticated",
-  # "all-unauthenticated", "cluster-authenticated", "cluster-unauthenticated", "deny"
-  # @default -- "all-unauthenticated"
-  defaultAllowPolicy: "all-unauthenticated"
```
Comment (mcharriere) [inline helm/linkerd2-app/values.yaml:78]: review this change

Comment (mcharriere) [reply helm/linkerd2-app/values.yaml:78]: it was renamed to defaultInboundPolicy

File: helm/linkerd2-app/values.yaml [hunk 4]
```diff
@@ -162,17 +175,26 @@ proxy:
...
+  # @default -- "all-unauthenticated"
+  defaultInboundPolicy: "all-unauthenticated"
 
 # proxy-init configuration
 proxyInit:
+  # -- Variant of iptables that will be used to configure routing. Currently,
+  # proxy-init can be run either in 'nft' or in 'legacy' mode. The mode will
+  # control which utility binary will be called. The host must support
+  # whichever mode will be used
+  iptablesMode: "legacy"
```
Comment (mcharriere) [inline helm/linkerd2-app/values.yaml:191]: check options and test them

**Comment by mcharriere** [comment]
review `Step 4: After deployment` section of readme and remove the unneeded parts

File: helm/linkerd2-app/values.yaml (hunk 1 above)
Comment (mcharriere) [reply helm/linkerd2-app/values.yaml]: done

File: helm/linkerd2-app/values.yaml [hunk 5]
```diff
@@ -220,10 +249,9 @@ enableH2Upgrade: true
 enablePSP: true
 
 # -- Failure policy for the proxy injector
-webhookFailurePolicy: Fail
+webhookFailurePolicy: Ignore
```
[the remark this answers is not in the document]
Comment (mcharriere) [reply helm/linkerd2-app/values.yaml]: reverted

**Comment by mcharriere** [comment]
I applied all the changed we spotted in the review meeting. I think now it's ready for review :).

File: helm/linkerd2-app/charts/linkerd-crds/templates/serviceprofile.yaml [hunk 6]
```diff
@@ -9,8 +9,8 @@ metadata:
   annotations:
     {{ include "partials.annotations.created-by" . }}
   labels:
-    linkerd.io/control-plane-ns: {{.Values.namespace}}
-    {{- include "linkerd.giantswarm-labels" . | nindent 4 }}
```
Comment (ced0ps) [inline helm/linkerd2-app/charts/linkerd-crds/templates/serviceprofile.yaml:13]: what your reason for removing the "giantswarm-labels". Labeling all the components helps people on support and in some cases is required for observability.

**Review by ced0ps — APPROVED** [review]
1 Question but LGTM
Hope I tracked everything correctly in the values.yaml, quite a few changes there.

File: helm/linkerd2-app/charts/linkerd-crds/templates/serviceprofile.yaml (hunk 6 above)
Comment (mcharriere) [reply helm/linkerd2-app/charts/linkerd-crds/templates/serviceprofile.yaml:13]: nice catch. I need to fix it. with the charts rename from upstream I missed it. thanks

Comment (mcharriere) [reply helm/linkerd2-app/charts/linkerd-crds/templates/serviceprofile.yaml:13]: fixed

**Pull request merged**